WIP: Update layout and routing

Tejasgarg opened the pull request:
This is a work in progress, and I will be updating this with the complete routing.

This PR aims to update the layout and routing for the board with updates from the Oscilloscope front end and power circuitry.

- fixes: #49

CloudyPadmal reviewed (changes requested):
Why are there several resistors such as `R25`, `R22` etc are in 0805 package instead of 0603? Is that intentional or by mistake?

Also the variable capacitors have rather a wide courtyard. Make it smaller similar to stock footprints in KiCAD and place the capacitors (fixed+variable) close to each other.

I see some ground vias from components exit the pads diagonally, while some straight out. Is there a reason for the difference routing?

Did you check the impedance of the oscilloscope traces are matched to 50 Ohms? I see that almost all the traces are 0.2 mm. 

The orientation of switch labels `POWER` and `RESET` are different from one another. The labels aren't even center aligned with the switches. Please focus on the aesthetics and fix those issues.

Tejasgarg commented:
Ohh, on searching for it again 
6.12k and 10.2k resistors are available and close to the previous values.
This could also be a viable option
@CloudyPadmal @bessman thoughts?

bessman commented:
Regarding the layout, it mostly looks good except the ADC pin caps (C40/C41). They must be as close as possible to the actual pin on the MCU. The op-amp output resistors (R28/R29) should be kept close to the op-amp; the trace between R28/29 and C40/41 can be long-ish.

bessman commented:
The GDTs' ground should be bonded directly to the BNC shells, not via signal ground vias.

Tejasgarg commented:
> That is to reduce digital switching transient ground currents from contaminating the analog circuitry.
> 
> I'm not 100% sure if the old hardware has it, it might.

Hey, @CloudyPadmal 
I was thinking of how we can implement separate ground planes for digital and analog circuitry.
However, with our design, this seems quite complicated as many analog and digital components are quite closely placed.
Here is the rough segregations for the analog and digital parts, what are your thoughts on how I should implement seperate ground planes.
Also, let me know if I have missed anything or wrongly marked any component
Thanks!

J5 Connector:
<img width="1228" height="187" alt="25 09 25 20" src="https://github.com/user-attachments/assets/a5dc9f68-55b4-4c7d-ba28-628ac87ffc77" /></br>

Complete Board
<img width="1310" height="736" alt="25 09 25 17" src="https://github.com/user-attachments/assets/b72976c3-59bf-4ea4-9ed8-c3468fa2bbab" />

CloudyPadmal commented:
Hi @Tejasgarg

The most sensitive analog circuit that we need to protect from transients is at the top right. So, we can have a ground pour that covers that region you have marked in a yellow square and have a small connection or two to connect with the rest of the ground plane. 

What do you think?

bessman commented:
> I have a small question here, C29 is a decoupling capacitor for the MCU and is placed about 1.5mm from the MCU.
> Am I overlooking something here?

The location of C29 is fine for the MCU. The op-amps in the scope input need their own decoupling caps close to their power rails.

Tejasgarg commented:
> Always place the capacitor closer to the IC pins, not the resistor.
> 

Ok! I will keep this in mind. Thanks!
> The two channels are almost identical. Why not make the layout symmetric?

Yes, as pointed out by @bessman as well during the meet.
I will be updating the layout to make both channels more symmetric.

Tejasgarg commented:
> > I have a small question here, C29 is a decoupling capacitor for the MCU and is placed about 1.5mm from the MCU.
> > Am I overlooking something here?
> 
> The location of C29 is fine for the MCU. The op-amps in the scope input need their own decoupling caps close to their power rails.

Ok! understood, thanks :)
I will add 100nF capacitors to the power rails for the op-amps.

CloudyPadmal reviewed (changes requested):
There is no silkscreen around the components other than on the sides (have a look at the [silkscreen render](https://raw.githubusercontent.com/fossasia/pslab-mini-hardware/layout-previews/layout-F.SilkS-51:4101d44.png)). Therefore a diagonally exiting ground via makes no sense. They also increase the length of the ground traces compared to a directly exiting trace. 

Please remove all the diagonal ground traces and make them straight and place the vias closer to the pad (but not over the pad).

bessman commented:
> Now each power rail of the op amps has an individual 100nF decoupling capacitor.

Looks good!

CloudyPadmal reviewed (changes requested):
I just noticed that the switch labels are facing downwards while all the other silkscreen labels are predominantly facing upwards. Rotate `POWER` and `RESET` switch labels by 180 degrees and make them center aligned with the mid point of switches.

Tejasgarg commented:
> Please remove all the diagonal ground traces and make them straight and place the vias closer to the pad (but not over the pad).

Hi @CloudyPadmal!
I have changed all the ground traces to be straight now.
 
> I really think they're still too far away. With ~10 mm of trace between the ADC pin and the cap, the inductance will cause large sampling kickback which will increase the settling time and potentially cause crosstalk.

Hey @bessman,
I have tried reducing this to be 3-5 mm now.
Let me know your thoughts on this, thanks!

> I just noticed that the switch labels are facing downwards while all the other silkscreen labels are predominantly facing upwards. Rotate `POWER` and `RESET` switch labels by 180 degrees and make them center aligned with the mid point of switches.
> 
Rotated and center aligned these.

Also noticed that it was actually showing up fine on my KiCad
But for some reason it was not showing up correctly in the output pictures 
<img width="568" height="451" alt="Screenshot 2025-09-26 201446" src="https://github.com/user-attachments/assets/9c047f48-eba1-45e0-8860-3a03452c17a2" />
<img width="365" height="285" alt="Screenshot 2025-09-26 201548" src="https://github.com/user-attachments/assets/0e35ca09-29b3-48d4-93ef-10ff71bdf103" />
Earlier, I was using Century Gothic for the font. I have switched this to the default KiCAD font to mitigate the alignment issue.

> There is also a discontinuity at the mid of `CH2_IN` line. Please fix that too.

Fixed!
